### PR TITLE
admin user index search form

### PR DESCRIPTION
**Pull request by minhld1709**

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -34,6 +43,10 @@ def update
     end
   end
 
+  def destroy
+    User.find(params[:id]).destroy
+    redirect_to admin_users_path
+  end
```
Comment (NguyenHai2310) [inline app/controllers/admin/users_controller.rb]: break line

File: app/views/admin/users/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<h1><%= t "admin.user.index.banner" %></h1>
+<%= form_tag admin_users_path, :method => 'get' do %>
```
Comment (NguyenHai2310) [inline app/views/admin/users/index.html.erb]: " instead '

File: app/views/admin/users/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<h1><%= t "admin.user.index.banner" %></h1>
+<%= form_tag admin_users_path, :method => 'get' do %>    
+  <%= text_field_tag :search, params[:search] %>
+  <%= submit_tag "Search", :name => nil %>
```
Comment (NguyenHai2310) [inline app/views/admin/users/index.html.erb]: try name: nil

File: app/controllers/admin/users_controller.rb [hunk 4]
```diff
@@ -1,6 +1,15 @@
 class Admin::UsersController < ApplicationController
   before_action :authenticate_user!, :check_admin
 
+  def index
+    if params[:search].nil?
+      @users = User.all
+    else
+      @users = User.search params[:search]
+    end
+    @users = @users.paginate page: params[:page], per_page: 10
```
Comment (ThuBM) [inline app/controllers/admin/users_controller.rb]: 10 cho vao settings

File: app/views/admin/users/index.html.erb (hunk 3 above)
Comment (NguyenHai2310) [inline app/views/admin/users/index.html.erb]: i18n

File: app/models/user.rb [hunk 5]
```diff
@@ -7,6 +7,12 @@ class User < ActiveRecord::Base
   validates :birthday, presence: true
 
   mount_uploader :image, ImageUploader
+  scope :normal, ->{where role: "normal"}
```
Comment (phuong1492) [inline app/models/user.rb]: use gem Setting?

File: app/models/user.rb [hunk 6]
```diff
@@ -7,6 +7,12 @@ class User < ActiveRecord::Base
...
+
+  def self.search(search) 
+    where("name like ?", "%#{search}%")
+  end
+  
   def admin?
     role == "admin"
```
Comment (phuong1492) [inline app/models/user.rb]: use gem Setting

File: app/models/user.rb [hunk 7]
```diff
@@ -7,6 +7,12 @@ class User < ActiveRecord::Base
   validates :birthday, presence: true
 
   mount_uploader :image, ImageUploader
+  scope :normal, ->{where role: "normal"}
+
+  def self.search(search) 
+    where("name like ?", "%#{search}%")
```
Comment (phuong1492) [inline app/models/user.rb]: try using gem ransack?

File: app/views/admin/users/_user.html.erb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<tr>
+  <td><%= link_to user.name, admin_user_path(user) %></td>
+  <td><%= user.birthday %></td>
+  <td><%= link_to "", edit_admin_user_path(user),class: "glyphicon glyphicon-edit" %>
+    <%= link_to "", admin_user_path(user), method: :delete, data: {confirm: "confirm"}, class: "glyphicon glyphicon-trash" %>
```
Comment (phuong1492) [inline app/views/admin/users/_user.html.erb]: I18n

File: app/views/admin/users/index.html.erb (hunk 3 above)
Comment (phuong1492) [inline app/views/admin/users/index.html.erb]: I18n?

File: app/controllers/admin/users_controller.rb [hunk 9]
```diff
@@ -1,6 +1,15 @@
 class Admin::UsersController < ApplicationController
   before_action :authenticate_user!, :check_admin
 
+  def index
+    if params[:search].nil?
```
Comment (phuong1492) [inline app/controllers/admin/users_controller.rb]: remove "nil?"

File: app/controllers/admin/users_controller.rb [hunk 10]
```diff
@@ -34,6 +43,10 @@ def update
     end
   end
 
+  def destroy
+    User.find(params[:id]).destroy
+    redirect_to admin_users_path
```
Comment (phuong1492) [inline app/controllers/admin/users_controller.rb]: should have flash here and use if else on this funtion

File: app/controllers/admin/users_controller.rb (hunk 4 above)
Comment (minhld1709) [inline app/controllers/admin/users_controller.rb]: khong su dung rails_config

File: app/controllers/admin/users_controller.rb (hunk 9 above)
Comment (minhld1709) [inline app/controllers/admin/users_controller.rb]: if remove dont accept function

File: app/models/user.rb (hunk 5 above)
Comment (minhld1709) [inline app/models/user.rb]: no use gem setting

File: app/models/user.rb (hunk 6 above)
Comment (minhld1709) [inline app/models/user.rb]: no use gem setting

File: app/controllers/admin/users_controller.rb (hunk 9 above)
Comment (phuong1492) [inline app/controllers/admin/users_controller.rb]: try it:
if params[:search]
 @users = User.search(params[:search])
else
 @users = @users.paginate page: params[:page], per_page: 10
end

File: app/models/user.rb (hunk 7 above)
Comment (minhld1709) [inline app/models/user.rb]: pull tiep theo su dung ransack

**Pull request merged**